### ISSUE TITLE
Move configuration to .ini files

### ISSUE THREAD

**Issue by rafaljanicki** [YOURLS/YOURLS#1892]
Instead of .php files as .ini files are more universal and easier to maintain.

Context: I've forked your project and want to modify how config is fetched - instead of getting it from local filesystem, I'd like to add support for Amazon Web Services Beanstalk & S3. In order to do that, I need to pass location of file and SECRET/ACCESS KEY by parameters to the application, download the file by using those parameters and finally parse the file.
This solution is fully compatible with cloud orientation and imo should be incorporated into YOURLS v2 as the future is already here ;) Of course, this type of fetching configuration shouldn't be default at all.

**Comment by ozh** [comment]
The idea is interesting but will be left for plugin territory. PHP files are indeed not entirely practical to edit, but 
* they contain critical info and they're not readable via the browser ([example](http://ozh.in/user/config.php)) and we cannot reasonably assume everyone will configure their server to make config.ini unreadable
* at some point we'll have an installer and no one will ever need to edit config files any more

This said, this is a very cool idea for a plugin. If you try this path and find hooks that are missing to do so, feel free to propose new ones.

**Comment by rafaljanicki** [comment]
As I've checked, I don't think plugin will handle this situation as config file is strictly fetched from disk. In my application I did dirty workaround in loader where I check server attributes and if they exist, I loaded file from S3 and place it on disk.

Anyway, I'd work on that and do the hooks + probably change the configuration loading behaviour, but as you say - there is no sense as installer will come in place of file.

As for installer, it'd be good to keep in mind cloud users as they get more and more popular. If you dump configuration from GUI installer into file, unfortunately it won't work. One way or another, you'd have to either allow plugins to provide configuration or add support for S3.

I'm willing to help in this matter, but I don't want to hop in in the middle of the works :)
